### PR TITLE
Num.arange works with other Number datatypes, and .to_tensor method for arrays

### PR DESCRIPTION
**Pull request by Iainmon**
Works like so:
```crystal
r = Num.arange(-10, 10, 1.to_f64) # Tensor(Float64)
puts r

r = Num.arange 5.0, step: 0.1
puts r
```
```
Tensor([-10, -9 , -8 , -7 , -6 , -5 , -4 , -3 , -2 , -1 , 0  , 1  , 2  ,
        3  , 4  , 5  , 6  , 7  , 8  , 9  ])

Tensor([0  , 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1  , 1.1, 1.2,
        1.3, 1.4, 1.5, 1.6, 1.7, 1.8, 1.9, 2  , 2.1, 2.2, 2.3, 2.4, 2.5,
        2.6, 2.7, 2.8, 2.9, 3  , 3.1, 3.2, 3.3, 3.4, 3.5, 3.6, 3.7, 3.8,
        3.9, 4  , 4.1, 4.2, 4.3, 4.4, 4.5, 4.6, 4.7, 4.8, 4.9])
```

### CODE REVIEW

**Comment by Iainmon** [comment]
`.to_tensor` works like so:

```crystal
t = [1, 2, 3, 4, 5, 6].to_tensor
puts t

t = [[1, 2, 3], [4, 5, 6]].to_tensor
puts t
```

```
Tensor([1, 2, 3, 4, 5, 6])

Tensor([[1, 2, 3],
        [4, 5, 6]])
```

**Comment by Iainmon** [comment]
Pull requests are confusing. I meant to create 2 separate ones, but this works fine I think.

File: src/tensor/array.cr [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require "./tensor"
+
+class Array(T)
```
Comment (christopherzimmerman) [inline src/tensor/array.cr:3]: Can you move this to `src/extensions`, to be consistent with the number extensions?

Comment (Iainmon) [reply src/tensor/array.cr:3]: Absolutely! Should I do that, and just make another commit?

**Comment by Iainmon** [comment]
Changes made :)

**Pull request merged**